Products/ZenModel/migrate/addKibanaRouteInZproxyConf.py

```diff
@@ -0,0 +1,78 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import logging
+import re
+
+log = logging.getLogger("zen.migrate")
+
+import Migrate
+import servicemigration as sm
+sm.require("1.0.0")
+
+
+class AddKibanaRouteInZproxyConf(Migrate.Step):
+
+    version = Migrate.Version(5,1,9)
+
+    KIBANA_NEW_ROUTE_CONFIG = """location ^~ /api/controlplane/kibana {
+            set $http_ws true;
+            proxy_pass http://127.0.0.1:5601;
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection "upgrade";
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            rewrite /api/controlplane/kibana$ / break;
+            rewrite /api/controlplane/kibana/(.*)$ /$1 break;
+        }"""
+
+    CONFIG_FILENAME = "/opt/zenoss/zproxy/conf/zproxy-nginx.conf"
+
+    OLD_ROUTE_RE="(location\s*\^\~\s*\/logview\/)"
+
+    def _update_kibana_config(self, zproxy_config, old_route_string):
+        """
+        Find the old kibana route and replace it with the new one
+        @param old_route_string: old route string found in the config file
+        """
+        new_content = ""
+        start = zproxy_config.content.find(old_route_string)
+        end = zproxy_config.content.find("}", start)
+        new_content = zproxy_config.content[:end+1] + \
+                      self.KIBANA_NEW_ROUTE_CONFIG + \
```

Comment:
This is removing the 1.1.x route for kibana. We need to add the 1.2 route only if it has not already been added